src/main/java/com/epam/test_generator/entities/api/CaseTrait.java

```diff
@@ -0,0 +1,24 @@
+package com.epam.test_generator.entities.api;
+
+import com.epam.test_generator.entities.Step;
+import com.epam.test_generator.services.exceptions.BadRequestException;
+import java.util.List;
+import java.util.Objects;
+
+public interface CaseTrait {
+
+    List<Step> getSteps();
+
+
+    default Step hasStep(Step step) {
+        if (getSteps() == null || !getSteps().contains(step)) {
```

Comment:
Let's avoid defensive null checks inside our domain. We assume that data received from services is valid.